<launch>

	<!--  Supported Oem7 messages - refer to Oem7 user manual. -->
	<rosparam file="$(find novatel_oem7_driver)/config/oem7_msgs.yaml" command="load" ns="/novatel/oem7"/>	

   	<!-- Supported IMUs - refer to Oem7 user manual-->
	<rosparam file="$(find novatel_oem7_driver)/config/oem7_supported_imus.yaml" command="load" ns="/novatel/oem7"/>


	<!-- Nodelet Manager, hosting the nodelets comprising the driver --> 
        <node pkg="nodelet" type="nodelet" args="manager" name="driver"  ns= "/novatel/oem7" output="screen" />
	
	<!-- Nodelet interfacing with receiver to handle Oem7 Logs and publish ROS messages.  -->
	<node pkg="nodelet" type="nodelet" name="main" ns="/novatel/oem7/receivers"
	      args="load novatel_oem7_driver/Oem7MessageNodelet /novatel/oem7/driver" output="screen">

	    <!-- Message Handler Plugins -->
	    <rosparam file="$(find novatel_oem7_driver)/config/std_msg_handlers.yaml" />
	
	    <param name="oem7_msg_decoder"   value="Oem7MessageDecoder"    type="string" />

        <!-- All unknown fragments are published to oem7raw topic -->
        <param name="oem7_publish_unknown_oem7raw" value="true" type="bool" />
	</node>

	<!-- Nodelet sending receiver configuration commands. -->
	<node pkg="nodelet" type="nodelet" name="config" ns="/novatel/oem7/receivers/main"
	      args="load novatel_oem7_driver/Oem7ConfigNodelet /novatel/oem7/driver" output="screen">
	</node>
	
	
	<!-- Standard receiver initialization commands. -->
	<rosparam file="$(find novatel_oem7_driver)/config/std_init_commands.yaml" ns="/novatel/oem7/receivers/main"/>
	
	<!-- Standard Oem7 raw messages to publish -->
	<rosparam file="$(find novatel_oem7_driver)/config/std_oem7_raw_msgs.yaml" ns="/novatel/oem7/receivers/main"/> 
	
	<!-- Standard Messages / Topics to publish -->
	<rosparam file="$(find novatel_oem7_driver)/config/std_msg_topics.yaml" ns="/novatel/oem7/receivers/main"/> 
	
	<!--  
	Built-in Self Test
	
	1. Record all topics published by the driver
	2. Post-process the collected data to establish HZ, etc.
	-->    
	<arg name="oem7_bist" default="false" /> 	
	<group if="$(arg oem7_bist)" >
	    <!-- Record ROS messages for postprocessing -->
	    <node pkg="rosbag" type="record" name="oem7_bist_rosbag" 
		  args="record -O oem7_bist -e /gps/(.*) | /novatel/oem7/(.*) -x /novatel/oem7/oem7raw --size=1024 --duration 60.0" />
		      
	    <!--  Prevents the driver from exiting-->
	    <param name="/oem7_bist_analysis/duration" value="70.0" />
	    <test test-name="oem7_bist_analysis" pkg="novatel_oem7_driver" type="oem7_bist.py"
	 	      time-limit="120.0"
	 	      args="oem7_bist"
	 	      />
	</group>
	
	
</launch>

